src/bin/pg_autoctl/ipaddr.c

```diff
@@ -244,3 +302,226 @@ isTestEnv(void)
 	const char *socketDir = getenv("PG_REGRESS_SOCK_DIR");
 	return socketDir != NULL && socketDir[0] == '\0';
 }
+
+
+/*
+ * From /Users/dim/dev/PostgreSQL/postgresql/src/backend/libpq/hba.c
+ */
+static bool
+ipv4eq(struct sockaddr_in *a, struct sockaddr_in *b)
+{
+	return (a->sin_addr.s_addr == b->sin_addr.s_addr);
+}
+
+
+/*
+ * From /Users/dim/dev/PostgreSQL/postgresql/src/backend/libpq/hba.c
+ */
+static bool
+ipv6eq(struct sockaddr_in6 *a, struct sockaddr_in6 *b)
+{
+	int			i;
+
+	for (i = 0; i < 16; i++)
+		if (a->sin6_addr.s6_addr[i] != b->sin6_addr.s6_addr[i])
+			return false;
+
+	return true;
+}
+
+
+/*
+ * findHostnameLocalAddress does a reverse DNS lookup given an hostname
+ * (--nodename), and if the DNS lookup fails or doesn't return any local IP
+ * address, then returns false.
+ */
+bool
+findHostnameLocalAddress(const char *hostname, char *localIpAddress, int size)
+{
+	int error;
+    struct addrinfo *dns_lookup_addr;
+	struct addrinfo *dns_addr;
+	struct ifaddrs *ifaddr, *ifa;
```

Comment:
I find the `ifa` name a bit non-telling, it seems to be iterating over `ifaddr`, so maybe the first should be `ifaddrs` (plural) and we can use `ifaddr` for iteration. Although the type is already called the same, don't exactly know if such a collision is allowed

Reply:
Fixed.